### PR TITLE
fix: update download api

### PR DESCRIPTION
**Pull request by namchuai**
## Describe Your Changes

- Allow user to specify their desired `model_id`.
- Add a thread safe data structure for DownloadTaskQueue. Working on refactor DownloadService to use new DownloadTaskQueue to support parallel download.
- Clean up alias API.

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/cli/commands/model_list_cmd.cc [hunk 1]
```diff
@@ -17,8 +17,9 @@ using namespace tabulate;
 using Row_t =
     std::vector<variant<std::string, const char*, string_view, Table>>;
 
-void ModelListCmd::Exec(const std::string& host, int port, std::string filter,
-                        bool display_engine, bool display_version) {
+void ModelListCmd::Exec(const std::string& host, int port,
+                        const std::string& filter, bool display_engine,
```
Comment (vansangpfiev) [inline engine/cli/commands/model_list_cmd.cc:21]: Nice! I missed this in the previous PR.

**Review by vansangpfiev — APPROVED** [review]

**Comment by louis-jan** [comment]
This change broke Jan's cortex because it blocks custom engine path support. @vansangpfiev @namchuai

**Comment by dan-menlo** [comment]
> This change broke Jan's cortex because it blocks custom engine path support. @vansangpfiev @namchuai

@louis-jan @vansangpfiev @namchuai Can I understand this a bit more in detail - what does it mean by blocking custom engine path support? 

Does this mean we have to revert?

**Comment by louis-jan** [comment]
> > This change broke Jan's cortex because it blocks custom engine path support. @vansangpfiev @namchuai
> 
> @louis-jan @vansangpfiev @namchuai Can I understand this a bit more in detail - what does it mean by blocking custom engine path support?
> 
> Does this mean we have to revert?

Hey @dan-homebrew guys fixed it already. Applications embed Cortex, which bundles engines that aren't fixed at ~/cortex/engines since `engines install` won't be done in run-time. They could be `mac-arm/engines, mac-intel/engines, or windows/engines` or pre-packaged. And they can point to the corresponding engine at runtime via env or args.

**Pull request merged**